File: src/Server/Taskly.Application/Projects/Queries/GetProjectsShortInfo/ProjectShortInfoVm.cs
using Taskly.Domain;

namespace Taskly.Application.Projects
{
    public class ProjectShortInfoVm
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string? ShortName { get; set; }
        public string Company { get; set; }
        public bool IsExternal { get; set; }
        public bool IsOpened { get; set; }
        public string ProjectManager { get; set; }
        public string ChiefEngineer { get; set; }
        public DateTime Start { get; set; }
        public DateTime End { get; set; }
        public DateTime? CloseDate { get; set; }
        public string Customer { get; set; }
        public string Contract { get; set; }

        public static ProjectShortInfoVm FromProject(Project p)
        {
            return new ProjectShortInfoVm
            {
                Id = p.Id,
                ChiefEngineer = p.ChiefEngineer?.Name,
                CloseDate = p.CloseDate,
                Company = p.Company?.Name,
                Contract = p.Contract,
                Customer = p.Customer.Name,
                IsExternal = p.Type == ProjectType.External,
                End = p.End,
                IsOpened = p.IsOpened,
                Name = p.Name,
                ProjectManager = p.ProjectManager?.Name,
                ShortName = p.ShortName,
                Start = p.Start,
            };
        }
    }
}

File: src/Client/src/models/ProjectShortInfoVm.ts
import { dateAsShortStr } from "../common/dateFormatter"

export class ProjectShortInfoVm {
    id: string = ''
    name: string = ''
    shortName: string = ''
    company: string = ''
    isOpened: boolean = false
    isExternal: boolean = false
    projectManager: string = ''
    chiefEngineer: string = ''
    start: number = 0
    end: number = 0
    closeDate: number | null = 0
    сustomer: string = ''
    сontract: string = ''

    // calculated
    startAsStr: string = ''
    endAsStr: string = ''
    closeDateAsStr: string = ''

    public static init(arg: ProjectShortInfoVm) {
        arg.startAsStr =  dateAsShortStr(new Date(arg.start))
        arg.endAsStr = dateAsShortStr(new Date(arg.end))
        if (arg.closeDate) {
            arg.closeDateAsStr = dateAsShortStr(new Date(arg.closeDate))
        }
    }
}
